### ISSUE TITLE
[BUG] Full node needs a large amount of IO for what it does

### ISSUE THREAD

**Issue by Atemu** [Chia-Network/chia-blockchain#1603]
**Describe the bug**

Running a full node puts a relatively high random IO load on the disk with the .chia directory (on the order of hundreds of KiB/s) which is on an SSD in my case.  
Having it on the HDD I'm plotting on actually causes the blockchain to fail syncing because it's stuck waiting for IO. 

My reasonably well informed guess is that it's caused by blockchain db synchronisation.

**To Reproduce**

1. Run a full Node
2. Run `iostat 1`
3. Observe burst of IO to main disk every few seconds

**Expected behavior**

A full node should have to sync to disk at most every few minutes; a loss of even a few minutes of blockchain state is negligible.

**Screenshots**

**Desktop (please complete the following information):**

- OS: Linux
- OS Version/Flavor: NixOS 21.05
- CPU: Intel Celeron J4105

**Additional context**

Running inside a Docker container with the chia dir in a Docker volume backed by ZFS.

**Comment by hoffmang9** [comment]
We're pretty sure it is this issue which will be out in 1.0.2 shortly - https://github.com/Chia-Network/chia-blockchain/commit/39227a8513f705c979863e8f90e9e06313dd5e73

**Comment by Atemu** [comment]
The issue has been reduced in severity a good amount but it's still doing a lot more IO than I need it to AFAICT.

As I said, I really don't think the possibility of losing a few minutes of blockchain state is worth spending this many SSD erase cycles on; especially since Chia is supposed to be used by regular people with consumer hardware.  
Syncing more often than once a minute is probably wasteful.

`iostat` tells me that there's ~500KiB/s of write activity on my SSD on average. This is a bit skewed upwards since I'm also doing other things on the machine sometimes and I'm not sure how much time is averaged but the same is true for many other people I imagine and the order of magnitute is about right.  
This comes out to ~50GiB/day which is quite a lot to ask for when consumer SSDs are rated for like 20GB/day and aren't expected to hit that very often.

**Comment by mariano54** [comment]
This does seem like a lot, considering the DB is less than 100MB. There might be a bug somewhere

**Comment by hoffmang9** [comment]
What log level are you on when monitoring this?

**Comment by Atemu** [comment]
Default everything but I just found an error in my monitoring: The disk names changed and I was looking at one of the HDDs. *facepalm*
  
I'm not 100% sure what's writing to it (probably txg syncs?) to but it can't be the chia node as it only shows up in `iotop` with ~100KiB/s for a few seconds every 30s or so. This significantly better than what I reported but still a lot more frequent and intensive than expected/necessary.  
A rough extrapolation of that would still on the order of GiB/day. 

My DB's size:

```
(venv) root@SOTERIA:/# du -sh ~/.chia/mainnet/db/blockchain_v1_mainnet.sqlite
296M    /root/.chia/mainnet/db/blockchain_v1_mainnet.sqlite
(venv) root@SOTERIA:/# du --apparent-size -sh ~/.chia/mainnet/db/blockchain_v1_mainnet.sqlite
471M    /root/.chia/mainnet/db/blockchain_v1_mainnet.sqlite
```

(Docker container in hostnet mode, not actually running chia as root)
